src/neml2/solvers/NewtonWithLineSearch.cxx

```diff
@@ -97,13 +121,27 @@ NewtonWithLineSearch::linesearch(NonlinearSystem & system,
 
     auto stop = at::logical_or(nR2 <= crit, nR2 <= std::pow(atol, 2));
 
+    if (at::max(crit).item<Real>() < 0)
+      flag = true;
+
     if (at::all(stop).item<bool>())
+    {
+      check = true;
       break;
+    }
 
+    // alpha = alpha.batch_expand_as(stop).clone();
     alpha.batch_index_put_({at::logical_not(stop)},
                            alpha.batch_index({at::logical_not(stop)}) / _linesearch_sigma);
   }
 
+  if (flag)
+    neml_assert(check,
+                "Nonlinear Solver failed to converge: Line Search produces negative stopping "
+                "criteria, try with other "
+                "linesearch_type, increase linesearch_cutback "
+                "or reduce linesearch_stopping_criteria");
```

Comment:
Let's make this a warning, as line search does not have to be successful. Newton-Raphson could continue with a "failed" line search.